server/services/interventionsService.test.ts

```diff
@@ -1925,6 +1925,53 @@ pactWith({ consumer: 'Interventions UI', provider: 'Interventions Service' }, pr
       expect(appointment.attendance!.additionalAttendanceInformation).toEqual('Alex missed the bus')
     })
   })
+
+  describe('recordAppointmentBehaviour', () => {
+    it('returns an updated action plan appointment with the service user‘s behaviour', async () => {
+      await provider.addInteraction({
+        state:
+          'an action plan with ID 345059d4-1697-467b-8914-fedec9957279 exists and has 2 2-hour appointments already',
+        uponReceiving:
+          'a POST request to set the behaviour for session 2 on action plan with ID 345059d4-1697-467b-8914-fedec9957279',
+        withRequest: {
+          method: 'POST',
+          path: '/action-plan/345059d4-1697-467b-8914-fedec9957279/appointment/2/record-behaviour',
+          body: {
+            behaviourDescription: 'Alex was well behaved',
+            notifyProbationPractitioner: 'no',
+          },
+          headers: { Accept: 'application/json', Authorization: `Bearer ${token}` },
+        },
+        willRespondWith: {
+          status: 200,
+          body: Matchers.like({
+            sessionNumber: 2,
+            appointmentTime: '2021-05-13T13:30:00+01:00',
+            durationInMinutes: 60,
+            behaviour: {
```

Comment:
can this information be included in the attendance DTO? also do we need the notify pp boolean in the response DTO?

```    "attendance": {
        "attended": "late",
        "additionalAttendanceInformation": "charlie missed the bus",
        "behaviour": "charlie bit me"
    }```

Reply:
my thinking on this is that the behaviour is part and parcel of the session attendance, since it happened whilst the SU was attending the session. it seems strange to me to have it be it's own thing in the response.

Reply:
For me, I felt it would be more useful to have attendance as "did they attend?" and behaviour as its own thing to make it easier to update one separate entity per page.

Why would we not want the notify boolean in there? As it's only for sending a notification to the PP on the backend?

Reply:
For reference, here's what we've decided the DTO will look like:

```
sessionFeedback: {
  attendance: {
    attended: "late",
    additionalAttendanceInformation: null,
  },
  behaviour: {
    behaviourDescription: "bad",
    notifyProbationPractitioner: false,
  }
}
```